src/components/Edit.vue

```diff
@@ -0,0 +1,136 @@
+<script setup lang="ts">
+import { useAsyncState } from '@vueuse/core'
+import { computed, ref } from 'vue'
+import { useApi } from '../composables/useApi'
+import { useAuth } from '../composables/useAuth'
+import { useSessionToken } from '../composables/useSessionToken'
+import { Leaderboards } from '../lib/api/Leaderboards'
+import { UpdateLeaderboardRequest } from '../lib/api/data-contracts'
+
+const props = defineProps<{
+	id: number
+}>()
+
+const updateError = ref('')
+
+const token = useSessionToken()
+
+const leaderboards = new Leaderboards({
+	baseUrl: import.meta.env.VITE_BACKEND_URL
+})
+
+const {
+	state: board,
+	error,
+	isLoading,
+	execute
+} = useAsyncState(async () => {
+	// TODO: Add param in BE that allows also fetching deleted boards
+	const resp = await leaderboards.getLeaderboard(props.id)
+	return resp.data
+}, null)
+
+const updateRequest = computed<UpdateLeaderboardRequest>(() => ({
+	name: board.value?.name,
+	info: board.value?.info,
+	slug: board.value?.slug
+}))
+
+async function submit() {
+	useApi(
+		() =>
+			leaderboards.updateLeaderboard(
+				props.id,
+				updateRequest.value,
+				useAuth(token.value)
+			),
+		async (data) => {
+			console.log(await data.json())
+			execute()
+		},
+		(error) => {
+			updateError.value =
+				'Failed to update: ' + (error as Response).status.toString(10)
+		}
+	)
+}
+</script>
+
+<template>
+	<div class="container">
+		<div v-if="isLoading">Loading...</div>
+		<div v-else-if="error" class="error-container">
+			<p class="errorText">{{ error }}</p>
+			<button @click="execute()" class="button">Reload</button>
+		</div>
+
+		<div v-else class="main-content">
+			<RouterLink
+				class="back-link"
+				:to="{ name: 'leaderboardView', params: { id } }"
+				title="Changes will not be saved."
+				>&lt; Back</RouterLink
+			>
+
+			<p v-if="updateError" class="error-text">{{ updateError }}</p>
+
+			<form @submit.prevent="submit" class="edit-form">
+				<span>ID:</span>
+				<span>{{ board?.id }}</span>
+				<label for="name">Name:</label>
+				<input v-model="updateRequest.name" id="name" />
+				<label for="slug">Slug:</label>
+				<input v-model="updateRequest.slug" id="slug" />
+				<span>Created:</span>
+				<span>{{ board?.createdAt }}</span>
+				<span>Deleted:</span>
+				<span v-if="board?.deletedAt">{{ board?.deletedAt }}</span>
+				<span v-else class="dim">&lt;Not deleted&gt;</span>
+				<label for="info">Info:</label>
+				<textarea v-model="updateRequest.info" id="info" rows="5" />
+				<button>Submit</button>
+			</form>
```

Comment:
Had to switch to this over [the original solution](https://github.com/leaderboardsgg/leaderboard-admin/pull/23/files#diff-89e49417fdc92ae14def0d4333337e7dd865d3f717bffa86a5cb78f6c66b7d56R78) (nesting `<input>`s within `<tr>`s) because the linter doesn't like that

Reply:
Wow you're right